src/CoreBundle/Controller/ActivityController.php

```diff
@@ -405,10 +406,27 @@ public function subSegmentInfoAction($id, Account $account, Trackdata $trackdata
         $statistics->setTimesToAnalyze([30, 60, 120, 300, 600, 720, 1800, 3600, 7200]);
         $statistics->findSegments();
 
+        $Frontend = new \Frontend(false, $this->get('security.token_storage'));
+        $context = new Context($id, $account->getId());
+        $route = $context->route();
```

Comment:
What I did not think about so far: I guess we need to catch activities without route (or with route but without geohashes), e.g. treadmill runs, which do have a distance array but not necessarily gps data.